src/services/walletConnect.ts

```diff
@@ -0,0 +1,533 @@
+    (topic: string) => {
+      const connections = Object.values(walletKit?.getActiveSessions() || {});
+      return connections.find(
+        (connection) =>
+          connection.topic === topic || connection.pairingTopic === topic
+      );
+    },
+    [walletKit]
+  );
+
+  const connect = useCallback(
+    async (copiedUri: string) => {
+      if (!walletKit) {
+        try {
+          await initWalletKit();
+        } catch (e) {
+          showToast({
+            title: 'WalletConnect error',
+            subtitle:
+              'Something went wrong with WalletConnect, please try again.',
+          });
+        }
+      }
+
+      try {
+        setIsLoadingConnect(true);
+        const peerWalletConnect = await walletKit?.core.pairing.pair({
+          uri: copiedUri,
+        });
+
+        // Update activeSessions after pairing
+        const updatedSessions = walletKit?.getActiveSessions();
+        setActiveSessions(updatedSessions);
+
+        // Wait for the session to be fully registered
+        await new Promise((resolve) => {
+          setTimeout(resolve, 1000);
+        });
+
+        const session = getSessionFromTopic(peerWalletConnect?.topic || '');
+        if (session) {
+          showToast({
+            title: `${session.peer.metadata.name}`,
+            subtitle: 'Connected via WalletConnect.',
+            image: `${session.peer.metadata.icons[0]}`,
+          });
+        }
+      } catch (e) {
+        if (`${e}`.includes('Missing or invalid')) {
+          showToast({
+            title: 'Missing or invalid connection',
+            subtitle: 'Missing or invalid WalletConnect connection.',
+          });
+        } else if (`${e}`.includes('Pairing already exists')) {
+          showToast({
+            title: 'Connection already exists',
+            subtitle: 'Please try again with a new WalletConnect connection.',
+          });
+        } else if (`${e}`.includes('URI has expired')) {
+          showToast({
+            title: 'Connection has expired',
+            subtitle: 'Please try again with a new WalletConnect connection.',
+          });
+        } else {
+          showToast({
+            title: 'Something went wrong.',
+            subtitle:
+              'Please make sure you are using a valid WalletConnect connection.',
+          });
+        }
+      }
+      setIsLoadingConnect(false);
+    },
+    [walletKit, initWalletKit, showToast, getSessionFromTopic]
+  );
+
+  const disconnect = useCallback(
+    async (topic: string) => {
+      if (!walletKit) {
+        try {
+          await initWalletKit();
+        } catch (e) {
+          showToast({
+            title: 'WalletConnect error',
+            subtitle:
+              'Something went wrong with WalletConnect, please try again.',
+          });
+        }
+      }
+
+      const sessionData = getSessionFromTopic(topic);
+
+      const dAppName = sessionData?.peer.metadata.name ?? 'dApp';
```

Comment:
Might be worth defensively coding as you did above:

```suggestion
      const dAppName = sessionData?.peer?.metadata?.name ?? 'dApp';
```

Reply:
You are right, done